woocommerce/woocommerce-android
# [Coupons] Coupon Details Menu UI

Part of: #5533 


### Description


This adds the Compose-based Top App Bar and Dropdown Menu on the Coupon Details screen. It also replaces more references from "coupon summary" to "coupon details".

This does not add functionality for sharing or copying the coupon inside the dropdown menu. It will be added on a separate PR.

### Discussion.

The iOS design of the menu shows a bottom dialog list. In this PR, I opted to use a top-right drop down design instead. This is similar to an existing pattern like on the Product Details screen.

@hichamboushaba Thanks for sharing about ways to set up and modify the nav bar in the fragment. In the end I decided to go ahead and make a Compose-based app bar, for these reasons:
1. It can be a good example pattern if in the future we want to add more and more Compose-based screen,
2. I find the Compose code easier to follow and to work with as everything is in one place, and they're relatively faster to add as well.
3. I also ran into the problem of duplicate top nav bars when trying to modify it in the fragment, and to not spend too much time into it I chose to get back to work on the Compose person.

What do you think?

### Testing instructions


1. Start app, go to Menu, then Coupons.
5. Select a coupon, this will load the Coupon Details screen.
6. Check and ensure the top app bar is shown.
7. Tap the triple-dot menu on top right and make sure two menu items are shown: "Copy Coupon Code" and "Share Coupon".
8. Make sure the dropdown can be opened and closed properly.
9. Tap the back arrow on the app bar and make sure it works as well.

### Images/gif


<img src="https://user-images.githubusercontent.com/266376/161704677-75701075-9a4d-416d-938c-b5ea450145f3.png" width="42%">
